src/Compilers/Core/Portable/SourceGeneration/GeneratorState.cs

```diff
@@ -10,21 +10,21 @@ namespace Microsoft.CodeAnalysis
     internal readonly struct GeneratorState
     {
         public GeneratorState(GeneratorInfo info)
-            : this(info, ImmutableArray<GeneratedSourceText>.Empty)
+            : this(info, ImmutableDictionary<GeneratedSourceText, SyntaxTree>.Empty)
         {
         }
 
-        private GeneratorState(GeneratorInfo info, ImmutableArray<GeneratedSourceText> sources)
+        private GeneratorState(GeneratorInfo info, ImmutableDictionary<GeneratedSourceText, SyntaxTree> sources)
         {
             this.Sources = sources;
             this.Info = info;
         }
 
-        internal ImmutableArray<GeneratedSourceText> Sources { get; }
+        internal ImmutableDictionary<GeneratedSourceText, SyntaxTree> Sources { get; }
```

Comment:
>ImmutableDictionary [](start = 17, length = 19)

Are we using this as a dictionary? If not, consider storing as `ImmutableArray<(GeneratedSourceText, SyntaxTree)>`.

Reply:
Annoyingly we are in one place. I think it should be possible to remove that usage, but will require a bit of a refactor. I'm going to make an issue to track that and come back to it because I think it's a much better way to do it.

---
In reply to: [395320188](https://github.com/dotnet/roslyn/pull/42507#discussion_r395320188) [](ancestors = 395320188)

Reply:
Filed [https://github.com/dotnet/roslyn/issues/42593](https://github.com/dotnet/roslyn/issues/42593)

---
In reply to: [395342426](https://github.com/dotnet/roslyn/pull/42507#discussion_r395342426) [](ancestors = 395342426,395320188)